Linting swift code does not work. Error: No lintable files found at paths: '/tmp/lint'

**Describe the bug**

Linting swift code does not work. The error appears:
```
...
Status: Downloaded newer image for norionomura/swiftlint:latest
The `swiftlint autocorrect` command is no longer available.
Please use `swiftlint --fix` instead.
Error: No lintable files found at paths: '/tmp/lint'
Correcting Swift files at paths /tmp/lint
```

**To Reproduce**
Steps to reproduce the behavior:

1. Follow the quick start CLI to add mega-linter to the iOS Swift project
2. Accept default values (except one case where you choose that it's a Swift project)
3. Push to the repo, create a PR to trigger the GitHub Action
4. See swift-lint error (not related to the codebase)

**Expected behavior**

- `swift-lint` starts linting the code base.

**Screenshots**

<img width="834" alt="Screenshot 2021-04-29 at 13 26 35" src="https://user-images.githubusercontent.com/505054/116544147-0d605800-a8ef-11eb-9c4d-d2ba95158a38.png">

- Contents of `ERROR-SWIFT_SWIFTLINT.log`
```
Results of swiftlint linter (version 0.43.1)
See documentation on https://nvuillam.github.io/mega-linter/descriptors/swift_swiftlint/
-----------------------------------------------

❌ [ERROR] for workspace /github/workspace
Linter raw log:
Unable to find image 'norionomura/swiftlint:latest' locally
latest: Pulling from norionomura/swiftlint
34667c7e4631: Pulling fs layer
d18d76a881a4: Pulling fs layer
119c7358fbfc: Pulling fs layer
2aaf13f3eff0: Pulling fs layer
969975223755: Pulling fs layer
1ea85296b57d: Pulling fs layer
178f29677290: Pulling fs layer
5f015d65fccd: Pulling fs layer
2aaf13f3eff0: Waiting
969975223755: Waiting
1ea85296b57d: Waiting
178f29677290: Waiting
5f015d65fccd: Waiting
d18d76a881a4: Verifying Checksum
d18d76a881a4: Download complete
119c7358fbfc: Verifying Checksum
119c7358fbfc: Download complete
2aaf13f3eff0: Verifying Checksum
2aaf13f3eff0: Download complete
1ea85296b57d: Verifying Checksum
1ea85296b57d: Download complete
34667c7e4631: Verifying Checksum
34667c7e4631: Download complete
5f015d65fccd: Verifying Checksum
5f015d65fccd: Download complete
969975223755: Verifying Checksum
969975223755: Download complete
34667c7e4631: Pull complete
178f29677290: Verifying Checksum
178f29677290: Download complete
d18d76a881a4: Pull complete
119c7358fbfc: Pull complete
2aaf13f3eff0: Pull complete
969975223755: Pull complete
1ea85296b57d: Pull complete
178f29677290: Pull complete
5f015d65fccd: Pull complete
Digest: sha256:2b118a4b74998159c4aa3cd3ccf6d73074bec6ad70cfab0be28f4459754a0640
Status: Downloaded newer image for norionomura/swiftlint:latest
The `swiftlint autocorrect` command is no longer available.
Please use `swiftlint --fix` instead.
Error: No lintable files found at paths: '/tmp/lint'
Correcting Swift files at paths /tmp/lint
```

- Contents of `.swiftlint.yml`
```
excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Pods

disabled_rules: # rule identifiers to exclude from running
  - force_cast
  - force_try # Remove when done
  - type_name
  - identifier_name
  - type_body_length
  - file_length

cyclomatic_complexity:
  ignores_case_statements: true
```

Thanks a lot! I tested, changed the action invocation in the workflow to use the latest tag like in the screenshot below.

<img width="468" alt="Screenshot 2021-05-17 at 11 58 28" src="https://user-images.githubusercontent.com/505054/118470701-35c9be00-b707-11eb-81f8-675caad25df6.png">

However, I'm still getting the same error `ErrCorrecting Swift files at paths /tmp/lint or: No lintable files found at paths: '/tmp/lint'`. Am I testing it correctly? I'm using the swift flavor.

Contents of `ERROR-SWIFT_SWIFTLINT.log`:
```
Results of swiftlint linter (version 0.43.1)
See documentation on https://nvuillam.github.io/mega-linter/descriptors/swift_swiftlint/
-----------------------------------------------

❌ [ERROR] for workspace /github/workspace
Linter raw log:
Unable to find image 'norionomura/swiftlint:latest' locally
latest: Pulling from norionomura/swiftlint
34667c7e4631: Pulling fs layer
d18d76a881a4: Pulling fs layer
119c7358fbfc: Pulling fs layer
2aaf13f3eff0: Pulling fs layer
969975223755: Pulling fs layer
1ea85296b57d: Pulling fs layer
178f29677290: Pulling fs layer
5f015d65fccd: Pulling fs layer
969975223755: Waiting
1ea85296b57d: Waiting
178f29677290: Waiting
5f015d65fccd: Waiting
2aaf13f3eff0: Waiting
119c7358fbfc: Verifying Checksum
119c7358fbfc: Download complete
d18d76a881a4: Verifying Checksum
d18d76a881a4: Download complete
2aaf13f3eff0: Verifying Checksum
2aaf13f3eff0: Download complete
1ea85296b57d: Verifying Checksum
1ea85296b57d: Download complete
34667c7e4631: Verifying Checksum
34667c7e4631: Download complete
5f015d65fccd: Verifying Checksum
5f015d65fccd: Download complete
969975223755: Verifying Checksum
969975223755: Download complete
34667c7e4631: Pull complete
178f29677290: Verifying Checksum
178f29677290: Download complete
d18d76a881a4: Pull complete
119c7358fbfc: Pull complete
2aaf13f3eff0: Pull complete
969975223755: Pull complete
1ea85296b57d: Pull complete
178f29677290: Pull complete
5f015d65fccd: Pull complete
Digest: sha256:2b118a4b74998159c4aa3cd3ccf6d73074bec6ad70cfab0be28f4459754a0640
Status: Downloaded newer image for norionomura/swiftlint:latest
ErrCorrecting Swift files at paths /tmp/lint
or: No lintable files found at paths: '/tmp/lint'
```

--- a/.cspell.json
+++ b/.cspell.json
@@ -992,6 +992,7 @@
         "linelength",
         "linesep",
         "linkcheck",
+        "lintable",
         "linterkey",
         "linterrc",
         "linterversion",

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ Note: Can be used with `oxsecurity/megalinter@beta` in your GitHub Action mega-l
       Learn more about context on [GH-4328](https://github.com/oxsecurity/megalinter/issues/4328)
 
 - Fixes
+  - [swiftlint](https://github.com/realm/SwiftLint) Fix swiftlint error where linter is unable to find lintable files. Fixes [#440](https://github.com/oxsecurity/megalinter/issues/440).
 
 - Reporters
 

--- a/docs/descriptors/swift_swiftlint.md
+++ b/docs/descriptors/swift_swiftlint.md
@@ -11,8 +11,8 @@ description: How to use swiftlint (configure, ignore files, ignore errors, help 
 
 - Version in MegaLinter: **0.57.1**
 - Visit [Official Web Site](https://github.com/realm/SwiftLint#readme){target=_blank}
-- Docker image: [norionomura/swiftlint:latest](https://hub.docker.com/r/norionomura/swiftlint){target=_blank}
-  - arguments: `-v {{WORKSPACE}}:/tmp/lint:rw`
+- Docker image: [ghcr.io/realm/swiftlint:latest](https://hub.docker.com/r/ghcr.io/realm/swiftlint){target=_blank}
+  - arguments: `-v {{WORKSPACE}}:/tmp/lint:rw -w /tmp/lint`
 - See [How to configure swiftlint rules](https://github.com/realm/SwiftLint#configuration){target=_blank}
 - See [How to disable swiftlint rules in files](https://github.com/realm/SwiftLint#disable-rules-in-code){target=_blank}
 - See [Index of problems detected by swiftlint](https://realm.github.io/SwiftLint/rule-directory.html){target=_blank}
@@ -82,11 +82,11 @@ swiftlint is called once on the whole project directory (`project` CLI lint mode
 ### Example calls
 
 ```shell
-docker run -v /tmp/lint:/tmp/lint:rw norionomura/swiftlint:latest swiftlint lint --path /tmp/lint --strict
+docker run -v /tmp/lint:/tmp/lint:rw ghcr.io/realm/swiftlint:latest swiftlint --strict
 ```
 
 ```shell
-docker run -v /tmp/lint:/tmp/lint:rw norionomura/swiftlint:latest swiftlint --fix --path /tmp/lint --strict
+docker run -v /tmp/lint:/tmp/lint:rw ghcr.io/realm/swiftlint:latest swiftlint --fix --strict
 ```
 
 

--- a/megalinter/descriptors/swift.megalinter-descriptor.yml
+++ b/megalinter/descriptors/swift.megalinter-descriptor.yml
@@ -19,15 +19,16 @@ linters:
     cli_lint_extra_args:
       - "lint"
     cli_lint_extra_args_after:
-      - "/tmp/lint"
       - "--strict"
     cli_lint_fix_arg_name: "--fix"
     cli_lint_fix_remove_args:
       - "lint"
-    cli_docker_image: norionomura/swiftlint
+    cli_docker_image: ghcr.io/realm/swiftlint
     cli_docker_args:
       - "-v"
       - "{{WORKSPACE}}:/tmp/lint:rw"
+      - "-w"
+      - "/tmp/lint"
     config_file_name: ".swiftlint.yml"
     cli_config_arg_name: ""
     cli_help_arg_name: "help"
@@ -35,8 +36,8 @@ linters:
     cli_lint_errors_count: regex_number
     cli_lint_errors_regex: "Found ([0-9]+) violations"
     examples:
-      - "docker run -v /tmp/lint:/tmp/lint:rw norionomura/swiftlint:latest swiftlint lint --path /tmp/lint --strict"
-      - "docker run -v /tmp/lint:/tmp/lint:rw norionomura/swiftlint:latest swiftlint --fix --path /tmp/lint --strict"
+      - "docker run -v /tmp/lint:/tmp/lint:rw ghcr.io/realm/swiftlint:latest swiftlint --strict"
+      - "docker run -v /tmp/lint:/tmp/lint:rw ghcr.io/realm/swiftlint:latest swiftlint --fix --strict"
     ide:
       atom:
         - name: linter-swiftlint

